http/src/main/scala/org/http4s/blaze/http/http1/server/Http1ServerCodec.scala

```diff
@@ -206,12 +206,14 @@ private final class Http1ServerCodec(maxNonBodyBytes: Int, pipeline: TailStage[B
     final override type Finished = RouteResult
 
     private[this] var closed = false
-    protected val lock: Object = this
+
+    final protected def lock: Object = this
 
     protected def doWrite(buffer: ByteBuffer): Future[Unit]
 
     protected def doFlush(): Future[Unit]
 
+    // Must be called from while holding the lock
```

Comment:
Not 100% clear on what you mean here.

Reply:
I tried to make it clearer, let me know if it's still a problem.